browser/lib/transport/jsonptransport.ts

```diff
@@ -0,0 +1,232 @@
+	* Kludge to still alow its responses to work, while not keeping the
+	* function form for normal use and not cluttering window.Ably
+	* https://github.com/expressjs/express/blob/5b4d4b4ab1324743534fbcd4709f4e75bb4b4e9d/lib/response.js#L305
+	*/
+_._ = function(id: string) { return _['_' + id] || noop; };
+let idCounter = 1;
+let head: HTMLHeadElement | null = null;
+const shortName = 'jsonp';
+
+let checksInProgress: Array<StandardCallback<boolean>> | null = null;
+
+if(Platform.jsonpSupported) {
+	head = document.getElementsByTagName('head')[0];
+}
+
+export function createRequest(uri: string, headers: Record<string, string> | null, params?: RequestParams, body?: unknown, requestMode?: number, timeouts?: Record<string, number> | null, method?: HttpMethods) {
+	/* JSONP requests are used either with the context being a realtime
+	 * transport, or with timeouts passed in (for when used by a rest client),
+	 * or completely standalone.  Use the appropriate timeouts in each case */
+	timeouts = timeouts || Defaults.TIMEOUTS;
+	return new Request(undefined, uri, headers, Utils.copy<RequestParams>(params), body, requestMode as number, timeouts as Record<string, number>, method);
+}
+
+class JSONPTransport extends CometTransport {
+	shortName = shortName;
+
+	constructor(connectionManager: ConnectionManager, auth: Auth, params: TransportParams) {
+		super(connectionManager, auth, params);
+		params.stream = false;
+	}
+
+	static isAvailable() {
+		return Platform.jsonpSupported && Platform.allowComet;
+	}
+
+	/* connectivity check; since this has a hard-coded callback id,
+	 * we just make sure that we handle concurrent requests (but the
+	 * connectionmanager should ensure this doesn't happen anyway */
+
+	static tryConnect(connectionManager: ConnectionManager, auth: Auth, params: TransportParams, callback: TryConnectCallback) {
+		const transport = new JSONPTransport(connectionManager, auth, params);
+		const errorCb = function(this: { event: string }, err: ErrorInfo) { callback({event: this.event, error: err}); };
+		transport.on(['failed', 'disconnected'], errorCb);
+		transport.on('preconnect', function() {
+			Logger.logAction(Logger.LOG_MINOR, 'JSONPTransport.tryConnect()', 'viable transport ' + transport);
+			transport.off(['failed', 'disconnected'], errorCb);
+			callback(null, transport);
+		});
+		transport.connect();
+	}
+
+	toString() {
+		return 'JSONPTransport; uri=' + this.baseUri + '; isConnected=' + this.isConnected;
+	}
+
+	createRequest(uri: string, headers: Record<string, string> | null, params?: Record<string, string>, body?: unknown, requestMode?: number, timeouts?: Record<string, number>, method?: HttpMethods) {
+		/* JSONP requests are used either with the context being a realtime
+		 * transport, or with timeouts passed in (for when used by a rest client),
+		 * or completely standalone.  Use the appropriate timeouts in each case */
+		timeouts = (this && this.timeouts) || timeouts || Defaults.TIMEOUTS;
```

Comment:
Can be simplified to this
```suggestion
		timeouts = this?.timeouts || timeouts || Defaults.TIMEOUTS;
```